xbmc/music/MusicDatabase.cpp

```diff
@@ -5877,3 +5894,39 @@ bool CMusicDatabase::GetFilter(CDbUrl &musicUrl, Filter &filter, SortDescription
 
   return true;
 }
+
+void CMusicDatabase::UpdateFileDateAdded(int songId, const std::string& strFileNameAndPath)
+{
+  if (songId < 0 || strFileNameAndPath.empty())
+    return;
+
+  CDateTime dateAdded;
+  try
+  {
+    if (NULL == m_pDB.get()) return;
+    if (NULL == m_pDS.get()) return;
+
+    // 1 prefering to use the files mtime(if it's valid) and only using the file's ctime if the mtime isn't valid
+    if (g_advancedSettings.m_iMusicLibraryDateAdded == 1)
+    {
+      dateAdded = CFileUtils::GetModificationDate(strFileNameAndPath);
+    }
+    //2 using the newer datetime of the file's mtime and ctime
+    else if (g_advancedSettings.m_iMusicLibraryDateAdded == 2)
+    {
+      dateAdded = CFileUtils::GetModificationDate(strFileNameAndPath, true);
+    }
+    //0 using the current datetime
+    else 
```

Comment:
could GetModificationDate returns an invalid date for a valid file?  it returns an invalid date for ab empty file. or should we exec the query only for a valid date?